src/clib/ct.cpp

```diff
@@ -284,6 +284,19 @@ extern "C" {
         }
     }
 
+    int thermo_getSpeciesCharges(int n, size_t lenm, double* sc)
+    {
+        try {
+            ThermoPhase& p = ThermoCabinet::item(n);
+            p.checkSpeciesArraySize(lenm);
+            const vector_fp& ch = p.speciesCharges();
+            copy(ch.begin(), ch.end(), sc);
```

Comment:
A simpler implementation of this would be simply
```c++
p.getSpeciesCharges(sc);
```

Reply:
No, that would replace both lines. `sc` is a pointer to an the array where you want to store the charges, and `Phase::getSpeciesCharges(double* charges)` does exactly that.

Reply:
Ah I see, that change is now up to date in this PR.